presto-main/src/main/java/com/facebook/presto/operator/MergeSort.java

```diff
@@ -0,0 +1,318 @@
+/*
```

Comment:
Do you need special custom implementation of `MergeSort`? Can't you use:http://google.github.io/guava/releases/16.0/api/docs/com/google/common/collect/Iterators.html#mergeSorted(java.lang.Iterable,%20java.util.Comparator)
?
You could convert `Iterator<Page>` into something like `Iterator<PageRow>` and back from `Iterator<PageRow>` to `Iterator<Page>` after sort.

`PageRow` could contain: `Page` reference and `position`